models/migrations/v68.go

```diff
@@ -29,75 +57,99 @@ func reformatAndRemoveIncorrectTopics(x *xorm.Engine) (err error) {
 	delTopicIDs := make([]int64, 0, batchSize)
 	ids := make([]int64, 0, 30)
 
+	log.Info("Validating existed topics...")
 	if err := sess.Begin(); err != nil {
 		return err
 	}
-	log.Info("Validating existed topics...")
 	for start := 0; ; start += batchSize {
 		topics = topics[:0]
-		if err := sess.Asc("id").Limit(batchSize, start).Find(&topics); err != nil {
+		if err := x.Cols("id", "name").Asc("id").Limit(batchSize, start).Find(&topics); err != nil {
 			return err
 		}
 		if len(topics) == 0 {
 			break
 		}
 		for _, topic := range topics {
-			if models.ValidateTopic(topic.Name) {
+			if validateTopic(topic.Name) {
 				continue
 			}
+			log.Info("Incorrect topic: id = %v, name = %q", topic.ID, topic.Name)
+
 			topic.Name = strings.Replace(strings.TrimSpace(strings.ToLower(topic.Name)), " ", "-", -1)
 
+			ids = ids[:0]
```

Comment:
It looks like you are resetting the `ids` variable here back to an empty array. Can you change this variable to a different scope, such as define `ids` inside this loop so you don't need to have this line?

Reply:
Changed all these assignments